app/controllers/api/v1/visualization/clusters_controller.rb

```diff
@@ -228,12 +228,13 @@ def self.get_cluster_viz_data(study, cluster, url_params)
           end
 
           if cluster.is_3d? && cluster.has_range?
-            aspect = ClusterVizService.compute_aspect_ratios(range)
+            aspect = ClusterVizService.compute_aspect_ratios(cluster.domain_ranges)
           end
 
           axes_full = {
             titles: titles,
-            aspects: aspect
+            aspects: aspect,
+            ranges: cluster.domain_ranges
```

Comment:
we're already sending the `domain_ranges` in the response as `userSpecifiedRanges`.  I think the bug is in get3DScatterProps() in `ScatterPlot.js`.  There we check for userSpecifiedRanges, but are still trying to read values from axes.ranges.

Alternatively, if you think sending them as part of the axes is clearer (I don't have strong feelings, except I'd prefer that something in the variable name should indicate they were set by the study owner, and aren't just auto-computed), feel free to remove that userSpecifiedRanges from the root object and update the frontend code in ScatterPlot.

Reply:
Ah - thanks for the clarification.  You're right in that it is being shipped twice... I just glossed over the `userSpecifiedRanges`.  I think it's fine to leave it in there since it already exists, and I can adjust to make sure we read/honor those, and update the tests accordingly.